### PR TITLE
Bump dependencies

### PR DESCRIPTION
**Pull request by lgoettgens**
This prepares Oscar for the breaking releases of AbstractAlgebra v0.32.0, Nemo v0.36.0, and Hecke v0.21.0.

cc @thofma @fingolfin

### CODE REVIEW

File: src/Rings/MPolyMap/MPolyRing.jl [hunk 1]
```diff
@@ -59,7 +59,7 @@ defined by
   x -> y
   y -> x
 with map on coefficients
-#1
+  #1
```
Comment (lgoettgens) [inline src/Rings/MPolyMap/MPolyRing.jl:62]: @simonbrandhorst do you have an idea what's happening here? The change is possibly due to https://github.com/Nemocas/AbstractAlgebra.jl/pull/1436, but the printing was weird before as well.

Comment (thofma) [reply src/Rings/MPolyMap/MPolyRing.jl:62]: Its an anonymous function

**Comment by lgoettgens** [comment]
The CI job reports:
```
Invalidations on default branch: 1823 (1823 via deps)
This branch: 1826 (1826 via deps)
```

This low increase seems to be okay for the large amount of stuff moved from Nemo to AA.

**Comment by thofma** [comment]
This is just https://github.com/Nemocas/AbstractAlgebra.jl/pull/1436

**Pull request merged**